app/concerns/koudoku/subscription.rb

```diff
@@ -95,13 +95,6 @@ def processing!
                 end
               end
 
-              # If the class we're being included in supports coupons ..
-              if respond_to? :coupon
-                if coupon.present? and coupon.free_trial?
-                  customer_attributes[:trial_end] = coupon.free_trial_ends.to_i
-                end
-              end
-
               customer_attributes[:coupon] = @coupon_code if @coupon_code
```

Comment:
This specific attribute  `coupon` seems to actually be either a customer or subscription attribute. So, I left where it is:

* https://stripe.com/docs/api/customers/create#create_customer-coupon
* https://stripe.com/docs/api/subscriptions/create#create_subscription-coupon